frontend/simple/views/Invite.vue

```diff
@@ -98,22 +102,32 @@ export default {
       this.members.splice(index, 1)
     },
     submit: async function () {
-      for (let i = 0; i < this.members.length; i++) {
-        let member = this.members[i]
-        let events = await backend.eventsSince(member.contractId, member.contractId)
-        let [contract, ...actions] = events.map(e => {
-          return Events[e.entry.type].fromObject(e.entry, e.hash)
-        })
-        let state = contract.toVuexState()
-        actions.forEach(action => {
-          let type = action.constructor.name
-          contract.constructor.vuex.mutations[type](state, action.data)
-        })
-        let mailbox = await backend.latestHash(state.attributes.mailbox)
-        let invite = new Events.PostInvite({groupId: this.$store.state.currentGroupId}, mailbox)
-        await backend.publishLogEntry(state.attributes.mailbox, invite)
+      try {
+        this.errorMsg = null
+        for (let i = 0; i < this.members.length; i++) {
+          let member = this.members[i]
+          let events = await backend.eventsSince(member.contractId, member.contractId)
+          let [contract, ...actions] = events.map(e => {
+            return Events[e.entry.type].fromObject(e.entry, e.hash)
+          })
+          let state = contract.toVuexState()
+          actions.forEach(action => {
+            let type = action.constructor.name
+            contract.constructor.vuex.mutations[type](state, action.data)
+          })
+          let mailbox = await backend.latestHash(state.attributes.mailbox)
+          let date = new Date()
+          let invite = new Events.PostInvite({groupId: this.$store.state.currentGroupId, sentDate: date.toString()}, mailbox)
+          await backend.publishLogEntry(state.attributes.mailbox, invite)
+          let latest = await backend.latestHash(this.$store.state.currentGroupId)
+          let invited = new Events.RecordInvitation({ username: member.name, inviteHash: invite.toHash(), sentDate: date.toString() }, latest)
+          await backend.publishLogEntry(this.$store.state.currentGroupId, invited)
```

Comment:
Another thing we should check for is that we handle the situation where there's a partial error, i.e. we invite half of the members but there's an error that prevents others from being invited.

I would add a TODO comment to address this, i.e. by marking somehow every member as invited or not, and maybe removing the ones that we successfully invited from the list.